### PR TITLE
Add update-components target, Fix PHONY Targets

### PR DESCRIPTION
**Pull request by seemethere**
This allows easier updating of Docker CE components

Signed-off-by: Eli Uriegas <eli.uriegas@docker.com>

### CODE REVIEW

**Comment by seemethere** [comment]
CC @andrewhsu @vieux @thaJeztah

**Comment by andrewhsu** [comment]
i'd rather have targets for each component:
```
$ make update-component-cli
$ make update-component-engine
$ make update-component-packaging
```

**Comment by seemethere** [comment]
@andrewhsu added in latest push

File: Makefile [hunk 1]
```diff
@@ -1,27 +1,55 @@
 CLI_DIR:=$(CURDIR)/components/cli
 ENGINE_DIR:=$(CURDIR)/components/engine
 PACKAGING_DIR:=$(CURDIR)/components/packaging
+MOBY_COMPONENTS_URL=https://raw.githubusercontent.com/shykes/moby-extras/master/cmd/moby-components
```
Comment (andrewhsu) [inline Makefile]: let's lock this so it is explicit what code is run: https://raw.githubusercontent.com/shykes/moby-extras/c54159187efebc6155821dc0493c2294e83203c1/cmd/moby-components

File: Makefile [hunk 2]
```diff
@@ -1,27 +1,55 @@
...
 	-$(MAKE) -C $(ENGINE_DIR) clean
 	-$(MAKE) -C $(PACKAGING_DIR) clean
+
+$(MOBY_COMPONENTS):
+	mkdir -p .helpers
+	curl -fsSL $(MOBY_COMPONENTS_URL) > $(MOBY_COMPONENTS)
+	chmod +x $(MOBY_COMPONENTS)
+
+.PHONY: update-components
+update-components: update-components-cli update-components-engine update-components-packaging
```
Comment (andrewhsu) [inline Makefile]: add comment so it shows up in `make help`: `## update components using moby extra tool`

File: Makefile (hunk 1 above)
Comment (seemethere) [reply Makefile]: Updated to include SHA as a variable to make it easier to change later on.

File: Makefile (hunk 2 above)
Comment (seemethere) [reply Makefile]: Added

**Comment by seemethere** [comment]
Made `MOBY_COMPONENTS` sha based so that if you change the sha of the script it'll automatically have to reload the newer version of the script.

**Pull request merged**